test/runtests.jl

```diff
@@ -1,10 +1,42 @@
 # Check that package loads
 using StatsKit
 
-using Pkg
+using Test
 
 # TODO: run all tests in the same process after 'using StatsKit'
 #       to check that they pass when all packages are loaded at the same time
-Pkg.test(["Bootstrap", "CategoricalArrays", "Clustering", "CSV", "DataFrames",
-          "Distances", "Distributions", "GLM", "HypothesisTests", "KernelDensity",
-          "Loess", "MultivariateStats", "MixedModels", "StatsBase", "ShiftedArrays", "TimeSeries"])
+#       This is not possible currently as Pkg.test starts a new environment
+
+# Check that reexported packages do not export different objects under the same name
+objects = Dict{Any, Symbol}()
+for m in [Bootstrap, CategoricalArrays, Clustering, CSV, DataFrames,
+          Distances, Distributions, GLM, HypothesisTests, KernelDensity,
+          Loess, MultivariateStats, MixedModels, StatsBase, ShiftedArrays, TimeSeries]
+    for name in names(m)
+        # Work around undefined exports:
+        # JuliaStats/StatsBase#663,
+        # JuliaStats/Clustering.j#210,
+        # JuliaStats/GLM.jl#405,
+        # JuliaStats/StatsModels.jl#218,
+        # JuliaStats/MultivariateStats.jl#142,
+        # JuliaArrays/ShiftedArrays.jl#42,
+        # JuliaStats/TimeSeries.jl#487
+        name in (:findat, :kmeans_opts, :Gaussian, :LagTerm, :dropterm,
+                 :invsqrtm!, :wmean!, :reduce_vec, :mapreduce_vec, :find) && continue
+        push!(objects, getproperty(m, name) => name)
+    end
+end
+
+exceptions = [:ci, :head, :lead, :predict, :predict, :rename!, :rename, :msd, :evaluate,
+              :rmsd, :ci, :rename!, :lag, :lag, :tail, :colwise, :colwise, :head, :meanad,
+              :msd, :transform, :tail, :rmsd, :rename, :transform, :evaluate, :lead]
```

Comment:
So it turns out that this test was more interesting than I thought! It would be worth running it on all JuliaStats and JuliaData packages. Conflicts seem to indicate that we should really create a StatsAPI package so that we override common definitions of standard methods.

See https://github.com/JuliaStats/StatsKit.jl/pull/22#issuecomment-783285090 about conflicts between ShiftedArrays and TimeSeries (since the addition of ShiftedArrays hasn't be released yet).

Conflicts between StatsBase and Distances should probably be resolved by deprecating the StatsBase functions in favor of those in Distances, since the latter is more specialized and more lightweight.

```julia
conflicts = Dict{Any,Symbol}(TimeSeries.rename => :rename,DataFrames.rename => :rename)
conflicts = Dict{Any,Symbol}(ShiftedArrays.lead => :lead,TimeSeries.lead => :lead)
conflicts = Dict{Any,Symbol}(Loess.predict => :predict,StatsBase.predict => :predict)
conflicts = Dict{Any,Symbol}(HypothesisTests.ci => :ci,Bootstrap.ci => :ci)
conflicts = Dict{Any,Symbol}(StatsBase.meanad => :meanad,Distances.meanad => :meanad)
conflicts = Dict{Any,Symbol}(HypothesisTests.ci => :ci,Bootstrap.ci => :ci)
conflicts = Dict{Any,Symbol}(Distances.evaluate => :evaluate,MultivariateStats.evaluate => :evaluate)
conflicts = Dict{Any,Symbol}(ShiftedArrays.lag => :lag,TimeSeries.lag => :lag)
conflicts = Dict{Any,Symbol}(StatsBase.msd => :msd,Distances.msd => :msd)
conflicts = Dict{Any,Symbol}(StatsBase.msd => :msd,Distances.msd => :msd)
conflicts = Dict{Any,Symbol}(StatsBase.rmsd => :rmsd,Distances.rmsd => :rmsd)
conflicts = Dict{Any,Symbol}(TimeSeries.rename => :rename,DataFrames.rename => :rename)
conflicts = Dict{Any,Symbol}(ShiftedArrays.lag => :lag,TimeSeries.lag => :lag)
conflicts = Dict{Any,Symbol}(TimeSeries.rename! => :rename!,DataFrames.rename! => :rename!)
conflicts = Dict{Any,Symbol}(Distances.evaluate => :evaluate,MultivariateStats.evaluate => :evaluate)
conflicts = Dict{Any,Symbol}(StatsBase.meanad => :meanad,Distances.meanad => :meanad)
conflicts = Dict{Any,Symbol}(DataFrames.transform => :transform,MultivariateStats.transform => :transform)
conflicts = Dict{Any,Symbol}(StatsBase.rmsd => :rmsd,Distances.rmsd => :rmsd)
conflicts = Dict{Any,Symbol}(DataFrames.transform => :transform,MultivariateStats.transform => :transform)
conflicts = Dict{Any,Symbol}(TimeSeries.rename! => :rename!,DataFrames.rename! => :rename!)
conflicts = Dict{Any,Symbol}(Loess.predict => :predict,StatsBase.predict => :predict)
conflicts = Dict{Any,Symbol}(ShiftedArrays.lead => :lead,TimeSeries.lead => :lead)
```

Reply:
Yes - this is super useful. StatsAPI.jl is needed. But I guess this should not delay merging this PR. Right?

Reply:
Yes, this PR ensures we catch new conflicts so better merge it.